rfcs/45-stageless.md

```diff
@@ -0,0 +1,564 @@
+  - e.g. `.add_system(physics.run_if(GameLogic::Physics))`
+
+System configuration can be stored in a `SystemConfig` struct. This can be useful to reuse, compose and quickly apply complex configuration strategies, before applying these strategies to labels or individual systems.
+
+If you just want to run your game logic systems in the middle of your schedule, after input is processed but before rendering occurs, add the premade `CoreLabel::GameLogic` to them.
+
+### Label configuration
+
+Each system label has its own associated `SystemConfig`, stored in the corresponding `Schedule`.
+This configuration is applied in an additive fashion to each system with that label.
+
+You can define labels in a standalone fashion, configuring them at the time of creation:
+
+```rust
+#[derive(SystemLabel)]
+enum Physics {
+    Forces,
+    CollisionDetection,
+    CollisionHandling,
+}
+
+impl Plugin for PhysicsPlugin{
+
+    fn build(app: &mut App){
+        // Within each frame, physics logic needs to occur after input handling, but before rendering
+        let mut common_physics_config = SystemConfig::new().after(CoreLabel::InputHandling).before(CoreLabel::Rendering);
+
+        app
+        // We can reuse this shared configuration on each of our labels
+        .add_label(Forces.configure(common_physics_config).before(Physics::CollisionDetection))
+        .add_label(Physics::CollisionDetection.configure(common_physics_config).before(Physics::CollisionHandling))
+        .add_label(Physics::CollisionHandling.configure(common_physics_config))
+        // And then apply that config to each of the systems that have this label
+        .add_system(gravity.label(Forces))
+        // These systems have a linear chain of ordering dependencies between them
+        // Systems earlier in the chain must run before those later in the chain
+        .add_system_chain([broad_pass, narrow_pass].label(Physics::CollisionDetection))
+        // System sets apply a set of labels to a collection of systems
+        // and are helpful for reducing boilerplate
+        .add_system_set([compute_forces, collision_damage], Physics::CollisionHandling);
+    }
+}
+```
+
+Or, you can apply labels to your systems, and then configure them later:
+
+```rust
+impl Plugin for PhysicsPlugin{
+
+   // This code has the exact same effect as the snipped above
+    fn build(app: &mut App){
+        let mut common_physics_config = SystemConfig::new().after(CoreLabel::InputHandling).before(CoreLabel::Rendering);
+
+        app
+        .add_system(gravity.label(Forces))
+        .add_system_chain([broad_pass, narrow_pass].label(Physics::CollisionDetection))
+        .add_system_set([compute_forces, collision_damage], Physics::CollisionHandling)
+        // This adds the new configuration constraints to the labels in an additive fashion
+        .configure_label(Physics::Forces.configure(common_physics_config).before(Physics::CollisionDetection))

```

Comment:
`add_label` and `configure_label` seem like they can be merged.

Reply:
Done.